lib/panoptes/client/authentication.rb

```diff
@@ -0,0 +1,73 @@
+require 'deprecate'
+
+module Panoptes
+  class Client
+    module Authentication
+      extend Gem::Deprecate
+      def jwt_payload
+        raise NotLoggedIn unless @auth[:token]
+        payload, = JWT.decode @auth[:token], jwt_signing_public_key, algorithm: 'RS512'
+        payload
+      rescue JWT::ExpiredSignature
+        raise AuthenticationExpired
+      end
+
+      def token_contents
+        jwt_payload.fetch('data')
+      end
+
+      def token_expiry
+        jwt_payload.fetch('exp')
+      end
+
+      def authenticated?
```

Comment:
as far as that goes, i just replicated the methods that existed in the client or were in caesar. i'm not sure what the reasons for distinguishing between these downstream

Reply:
Should we take the chance to refactor / clean up the code? 

I would remove all the `authenticated?` checks and `NotLoggedIn` raises and instead 500 here if there is a token that doesn't meet the schema as that should be automatically assigned as that means there is an integrity issue that we should probably be aware of.

Reply:
i think i'd prefer to keep the `authenticated?` signature rather than the `ok?` signature then, since it's a little more explicit. but given this is a library and not an app, i think raising an exception when not authenticated is the appropriate way to deal with it, because it leaves it up to the consumer to decide how to proceed

Reply:
We might want to consider adding a `NotAuthorized` exception in fact to go along with `AuthenticationExpired` and replace `NotLoggedIn` with `NotAuthenticated`

Reply:
I like `AuthenticationExpired` 💃 . What does `NotAuthorized` mean in this context? Can't decrypt the token?

Reply:
distinguishing between authentication and authorization, so like "we don't know who you are" vs. "you can't touch that"